mocked_correlator_api.py

```diff
@@ -0,0 +1,84 @@
+from mock import MagicMock
+from pvdb import Records
+
+import numpy as np
+from time import time
+
+elements_in_float_array = Records.CORRELATION_FUNCTION.value.database_entries["CORRELATION_FUNCTION"]["count"]
+
+DATA_USED_IN_IOC_SYSTEM_TESTS = np.linspace(0, elements_in_float_array, elements_in_float_array)
+
+
+class MockedCorrelatorAPI:
+    def __init__(self):
+        self.device = MagicMock()
+
+        self.device.start = MagicMock(side_effect=self.start)
+        self.device.configure = MagicMock(side_effect=self.configure)
+
+        self.device.MeasurementOn = MagicMock(side_effect=self.is_measurement_on)
+        self.device.update = MagicMock(side_effect=self.update)
+
+        self.device.measurement_on = False
```

Comment:
I don't think `measurement_on` needs to be on the device?

Reply:
We have to poll `self.device.MeasurementOn()` to determine whether or not the device is still taking a measurement, so I think this logic is correct